File: BrickBreakerADT/src/com/carlospinan/brickbreakerandroid/game/sprites/Block.java
package com.carlospinan.brickbreakerandroid.game.sprites;

import android.graphics.Bitmap;

import com.carlospinan.brickbreakerandroid.utils.Rectangle;
import com.carlospinan.brickbreakerandroid.utils.Sprite;

/**
 * Created by mac on 14/07/13.
 */
public class Block extends Sprite {

	public static final int STATE_ALIVE = 1;
	public static final int STATE_DESTROYED = 2;

	private int state;

	public Block(Bitmap bitmap) {
		super(bitmap);
		state = STATE_ALIVE;
	}

	public void destroy() {
		state = STATE_DESTROYED;
	}

	public int getState() {
		return state;
	}

	@Override
	public Rectangle getBounds() {
		float left = getX();
		float right = left + 10;
		float top = getY();
		float bottom = top + 10;
		return new Rectangle(left, top, right, bottom);
	}
}

File: BrickBreaker/app/src/main/java/com/carlospinan/brickbreaker/game/models/sprites/Block.kt
package com.carlospinan.brickbreaker.game.models.sprites

import android.graphics.Bitmap
import com.carlospinan.brickbreaker.game.models.Sprite

/**
 * @author Carlos Piñan
 */
enum class BlockState {
    ALIVE,
    DESTROYED
}

class Block(bitmap: Bitmap) : Sprite(bitmap) {

    private var _blockState = BlockState.ALIVE
    val blockState: BlockState
        get() = _blockState

    override fun update(dt: Float) {}

    fun destroy() {
        _blockState = BlockState.DESTROYED
    }

}